src/xdealer_channel.cpp

```diff
@@ -0,0 +1,51 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include "xeus-zmq/xmiddleware.hpp"
+
+#include "xdealer_channel.hpp"
+
+namespace xeus
+{
+
+    xdealer_channel::xdealer_channel(zmq::context_t& context,
+                                    const std::string& transport,
+                                    const std::string& ip,
+                                    const std::string& port)
+        : m_socket(context, zmq::socket_type::dealer)
+    {
+        m_socket.connect(get_end_point(transport, ip, port));
+    }
+
+    xdealer_channel::~xdealer_channel()
+    {
+    }
+
+    void xdealer_channel::send_message(zmq::multipart_t& message)
+    {
+        message.send(m_socket);
+    }
+
+    std::optional<zmq::multipart_t> xdealer_channel::receive_message(long timeout)
+    {
+        zmq::multipart_t wire_msg;
+        m_socket.set(zmq::sockopt::linger, static_cast<int>(timeout));
```

Comment:
Curious if a simple static_cast (long to int) would do the job here !

Reply:
Not sure to get what you mean; you *have* a `static_cast` here.